src/OrleansServiceFabricUtils/OrleansCommunicationListener.cs

```diff
@@ -42,7 +42,8 @@ public class OrleansCommunicationListener : ICommunicationListener
         /// </summary>
         /// <param name="context">The context.</param>
         /// <param name="configuration">The configuration.</param>
-        public OrleansCommunicationListener(ServiceContext context, ClusterConfiguration configuration)
+        /// <param name="siloName">The silo name, or <see langword="null"/> to have a name automatically selected.</param>
+        public OrleansCommunicationListener(ServiceContext context, ClusterConfiguration configuration, string siloName = null)
```

Comment:
Hm the `SiloName` you're talking about is in `ClusterConfiguration.Defaults`, I think. I believe that is not intended to be 'this' silo's name, it's just there because `Defaults` is also a `NodeConfiguration`, but `NodeConfiguration` also shouldn't need `SiloName`, since the `SiloName` is how you select the node configuration in the first place.

In the future, we can remove this notion of 'defaults' and 'overrides'. At that point, I think it does make sense to have the silo name in the config.